plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1021,6 +1021,10 @@ protected WriteMapping legacyToWriteMapping(@SuppressWarnings("unused") Connecto
     @Override
     public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<SortItem> sortOrder)
     {
+        topNFunction().ifPresent(topNFunction -> {
```

Comment:
Why exactly you want this?

Reply:
The default `supportsTopN` impl. is almost a no-op - but I think the tests will catch issues with an incorrectly implemented topN. I can remove this.

Reply:
i wanted to have this

Reply:
> i wanted to have this

:) It does not explain much. 

Is it because we have some default topNFunction. And we want to make sure that connector explicitly decides if it wants to use it or not?

Reply:
Correct. In other words, because we know the default implementation is responsibly correct only for the negative case.